Added homework lesson 17.

iBubbleGun opened the pull request:
Homework lesson 17 added.
Please, check it.

iBubbleGun committed:
Added homework lesson 17.

iBubbleGun committed:
Update homework lesson 17.
Minor code improvements.

kartius commented:
What for did you create a BallManager? If it related to the first task so you need to create your own concurrent list with several methods

kartius commented on src/main/java/edu/hillel/homework/lesson17/petrol_station/PetrolStation.java:
```diff
@@ -0,0 +1,75 @@
+    private static final int MAXIMUM_THREADS_ALLOWED = 3;
+    private final AtomicDouble amount;
+    private final ExecutorService executor;
+    private final CountDownLatch latch;
```
Nice to see that you use extra implementation of synchronizator
But!
Here it's not necessary. By the current logic, autos should wait till the last one starts to be refueling. But in real life, autos should make a refuel independently from each other

iBubbleGun committed:
Update homework lesson 17.
Removed waiting for all threads to complete before executing a method. Now the method is executed completely immediately and independently in different threads.
Also minor code improvements.

kartius commented on src/main/java/edu/hillel/homework/lesson17/petrol_station/PetrolStation.java:
```diff
@@ -0,0 +1,84 @@
+
+    public void doRefuel(Car car, double fuelAmount) {
+        executor.execute(() -> {
+            double newAmount = amount.get() - fuelAmount;
```
You can try
amount.updateAndGet(amount -> amount - fuelAmount);
And it will be wrapped operation by atomic function

iBubbleGun committed:
Update homework lesson 17.
Now use updateAndGet() atomic method instead set().
Also minor code improvements.

iBubbleGun committed:
Update homework lesson 17.
Minor code improvements.
